### ISSUE TITLE
make the !#x(x)!# part ditchable?

### ISSUE THREAD

**Issue by four2six** [allan-gam/GCal#23]
i'm sure you're not a blockly fan, but it would be neat if it could be used as well. 

for that to be applicable, the text device would have to be without the !#x(x)!# part, as blockly can't do any advanced comparing.

so it would be nice, if one could turn off the !#x(x)!# part for an imported calendar to use it in blockly or to have a fix name for comparison.

**Comment by four2six** [comment]
i found variables called gcal105ActiveEvents, etc
from the numbering i assume they are from your old "gcal" we used before and i obviously didn't deinstall completely. is that correct?

**Comment by allan-gam** [comment]
The GCal Calendar entry text device also contains some formatting. (Gray text and line break)  If blockly can not find a substring within a string it wouldn't help to just disable the !#x(x)!# part.

Are you trying to check what event that triggered? We can think about what would be the best way to solve this. Maybe putting the event name, the total number of  events, and the remaining number of  events in user variables. It's certainly possible to do that. Meanwhile maybe you can solve the problem using a filter on your GCal device?

The [BakSeeDaa´s Google Calendar for Domoticz](https://www.domoticz.com/forum/viewtopic.php?t=8333) has no uninstall script so if are not using it you'd have some Domoticz user variables named by the form GCalnnxxxXXXXXX that you can delete.

**Comment by four2six** [comment]
thanks, i will clear them to stop confusion. maybe we should add that to the wiki, if someone had installed the old one so they don't get confused. or maybe not. i have no clue how many installed the old gcal.

i was trying to check what the name of the actual event is, when the switch is on. not sure what i did when i tried, but it was definitely wrong. the name it gives me for the example below is "utilitydevice[191]". so the actual subject of this issue is not correct, as i can't access the name of the event in blockly at all, as far as i see it.

![bildschirmfoto 2017-04-30 um 11 27 09](https://cloud.githubusercontent.com/assets/4852434/25563249/17461a02-2d98-11e7-9aec-9a659a52feca.png)

the filter would work, but that would be really cumbersome.

in my case  i'd like to define something like this, with changing it depending on the day, the time of year, schedules, etc. i have some amazon dash buttons that are supposed to act differently according to what time the are used.

![bildschirmfoto 2017-04-30 um 11 30 17](https://cloud.githubusercontent.com/assets/4852434/25563267/8f7286b4-2d98-11e7-99de-438f870b4c12.png)

now if i had to do a gcal calendar for every of those defined areas, which would solve it, it would be really tough work and not very flexible. 
so in short: yes, user variables would be much better. and that would solve the whole issue.

**Comment by allan-gam** [comment]
I believe It's a good idea to write something in the Wiki about the old user variables that can be deleted. 

I will try to implement a user variable for each GCal entry that holds the name of the tripped event. That user variable will need a name so I can address it. I can not use the calendar address because there might be multiple GCal entries using different time deltas and/or keywords. The name of that user variable can not be altered later by the user.

**Comment by four2six** [comment]
give each calendar a unique number / ID at creation that your logic uses, but is never displayed anywhere and can only be assigned once.

**Comment by allan-gam** [comment]
I'll have a look into that.

**Comment by allan-gam** [comment]
The virtual text device information !#x(x)!# has been replaced by the more readable x(x). The purpose of the text device is to provide visual information in the GUI. Due to the formatting it will still not be suitable for scripting and we now have better solutions for that.

For scripting i recommend 
- Trigger on the change of the virtual switch device. Check the new state (On or Off)
- Check the newly introduced user variable GCal-XXX-trippedEvent for what event that tripped.
- Further scripting can be made using the user variables GCal-XXX-remainingEventsToday and GCal-XXX-eventsToday.

It would be nice to have a blockly example in the Wiki doing the above if you have the time for that.

Cheers!

**Comment by four2six** [comment]
reopened as a reminder to put all that in wiki.

**Comment by allan-gam** [comment]
Thanks

**Comment by four2six** [comment]
i guess the change hasn't been pushed yet? as i still see the !#x(x)!# and no user variables...

**Comment by allan-gam** [comment]
The latest version is 0.2.2. What version are you running?

**Comment by four2six** [comment]
0.2.2

was i supposed to delete the old calendars and set up new ones?

**Comment by allan-gam** [comment]
Yes, please have a look at https://github.com/allan-gam/GCal/blob/master/README.md

**Comment by four2six** [comment]
thanks, sorry for not reading that file ;-)

**Comment by four2six** [comment]
I get through the initial setup until:

```
Syncing Calendar data with Google: ... @import.calendar.google.com. Last fetched: 14400 minutes ago. Interval: 5
(u'2017-05-07T19:02:26Z', u'\u2193 21:04')
(u'2017-05-07T19:06:04Z', u'Civil twilight')
(u'2017-05-07T19:50:16Z', u'Nautical twilight')
(u'2017-05-07T20:43:31Z', u'Astronomical twilight')
(u'2017-05-08T00:56:25Z', u'Astronomical twilight')
(u'2017-05-08T02:10:53Z', u'Nautical twilight')
(u'2017-05-08T03:04:40Z', u'Civil twilight')
(u'2017-05-08T03:49:08Z', u'\u2191 5:51')
(u'2017-05-08T11:28:23Z', u'\u2600\ufe0f 13:28')
(u'2017-05-08T19:03:59Z', u'\u2193 21:06')
(u'2017-05-08T19:07:39Z', u'Civil twilight')
(u'2017-05-08T19:52:06Z', u'Nautical twilight')
(u'2017-05-08T20:45:53Z', u'Astronomical twilight')
(u'2017-05-09T00:51:56Z', u'Astronomical twilight')
(u'2017-05-09T02:08:25Z', u'Nautical twilight')
(u'2017-05-09T03:02:45Z', u'Civil twilight')
(u'2017-05-09T03:47:29Z', u'\u2191 5:49')
(u'2017-05-09T11:28:21Z', u'\u2600\ufe0f 13:28')
Unexpected response from Domoticz: ERR

```
switch and text device are created, plus 3 variables. the text device doesn't show the next event, maybe due to the zero minute thing or because UTF? the trippedEvent variable is 0, the other two variables are correct.

**Comment by allan-gam** [comment]
I have added some more debugging info. Can You please start all over again from the beginning with the new version I uploaded, 0.2.3? Thanks!

**Comment by allan-gam** [comment]
Please follow the instructions again in README.md

**Comment by four2six** [comment]
```
Syncing Calendar data with Google: ...@import.calendar.google.com. Last fetched: 14400 minutes ago. Interval: 5
(u'2017-05-07T19:02:26Z', u'\u2193 21:04')
(u'2017-05-07T19:06:04Z', u'Civil twilight')
(u'2017-05-07T19:50:16Z', u'Nautical twilight')
(u'2017-05-07T20:43:31Z', u'Astronomical twilight')
(u'2017-05-08T00:56:25Z', u'Astronomical twilight')
(u'2017-05-08T02:10:53Z', u'Nautical twilight')
(u'2017-05-08T03:04:40Z', u'Civil twilight')
(u'2017-05-08T03:49:08Z', u'\u2191 5:51')
(u'2017-05-08T11:28:23Z', u'\u2600\ufe0f 13:28')
(u'2017-05-08T19:03:59Z', u'\u2193 21:06')
(u'2017-05-08T19:07:39Z', u'Civil twilight')
(u'2017-05-08T19:52:06Z', u'Nautical twilight')
(u'2017-05-08T20:45:53Z', u'Astronomical twilight')
(u'2017-05-09T00:51:56Z', u'Astronomical twilight')
(u'2017-05-09T02:08:25Z', u'Nautical twilight')
(u'2017-05-09T03:02:45Z', u'Civil twilight')
(u'2017-05-09T03:47:29Z', u'\u2191 5:49')
(u'2017-05-09T11:28:21Z', u'\u2600\ufe0f 13:28')
Unexpected response from Domoticz: ERR Url: http://192.168.2.220:8080/json.htm?vtype=2&type=command&param=updateuservariable&vname=GCal-Sun+Events-trippedEvent&vvalue=
```

```
Debug is on
Traceback (most recent call last):
  File "/home/pi/GCal/gcal.py", line 1160, in <module>
    main(sys.argv[1:])
  File "/home/pi/GCal/gcal.py", line 1141, in main
    global googleCalStates; googleCalStates = loadGoogleCalStates()
  File "/home/pi/GCal/gcal.py", line 793, in loadGoogleCalStates
    createGoogleCalStateEntry(c)
  File "/home/pi/GCal/gcal.py", line 807, in createGoogleCalStateEntry
    googleCalStates.append(calStateEntry)
NameError: global name 'googleCalStates' is not defined
```

result unchanged

**Comment by allan-gam** [comment]
OK. I'm looking at it...

**Comment by allan-gam** [comment]
Can you show me the  /var/tmp/GCal/XXXXXXX@group.calendar.google.com.json file?
I'd like to see what kind of event caused this.

**Comment by allan-gam** [comment]
Regarding the 2nd error, I made an update. 0.2.4.

**Comment by four2six** [comment]
which part of the file do you need?

**Comment by allan-gam** [comment]
I would like to have the file here so I can test it. If it's a public calendar you can name it for me so I can test. If it contains sensitive data you can send me a PM https://www.domoticz.com/forum/ucp.php?i=pm&mode=compose&u=7064

Domoticz is throwing us an error and I suspect that the request URL has not been correctly encoded for characters like ☀️ and 🌧 etc. If you need to get it working immediately you may disable that particular calendar until I have fixed the problem.

**Comment by four2six** [comment]
not sure i want to put all that data on a server owned by someone

**Comment by allan-gam** [comment]
Isn't it a public calendar?

**Comment by four2six** [comment]
not really, it's public data, but a private calendar

**Comment by four2six** [comment]
this is how it begins, stripped of personal data, hopefully

```
[
  {
    "created": "2017-05-03T07:17:01.000Z",
    "creator": {
      "displayName": "http://www.webcal.fi/xxxververyverylong internetaddress
      "email": "xxx@import.calendar.google.com",
      "self": true
    },
    "description": "↓ 21:04\n\nSunset: 21:04\nChange: +1 min 34 s\n\nNight length: 8 h 46 min 43 s\nChange: -3 min 14 s",
    "end": {
      "dateTime": "2017-05-07T19:06:05Z"
    },
    "etag": "\"2987592643760000\"",
    "htmlLink": "https://www.google.com/calendar/event?eid=XzY5MTNjY3BoOGQwa2NiOXA4Y3BqZWI5azhncDM0YmExNjkyamdiYTI2bDBrOGg5aDhoMWs2Y3BsNmgwN2VwYjJjZGdtb2JqNmQ0IGMxM3AxcnZxbm5qaWtoan$
    "iCalUID": "longnumber@webcal.fi",
    "id": "xxxlong number",
    "kind": "calendar#event",
    "location": "Solingen",
    "organizer": {
      "displayName": "http://www.webcal.fi/xxxververyverylong internetaddress",
      "email": "xxx@import.calendar.google.com",
      "self": true
    },
    "reminders": {
      "useDefault": true
    },
    "sequence": 2680,
    "start": {
      "dateTime": "2017-05-07T19:02:26Z"
    },
    "status": "confirmed",
    "summary": "↓ 21:04",
    "transparency": "transparent",
    "updated": "2017-05-03T07:25:21.880Z",
    "visibility": "public"
  },
```

**Comment by allan-gam** [comment]
OK. If You send me the json for the following entries that you already disclosed above I will have enough data to work with.
```
(u'2017-05-07T19:02:26Z', u'\u2193 21:04')
(u'2017-05-07T19:06:04Z', u'Civil twilight')
(u'2017-05-07T19:50:16Z', u'Nautical twilight')
(u'2017-05-07T20:43:31Z', u'Astronomical twilight')
(u'2017-05-08T00:56:25Z', u'Astronomical twilight')
(u'2017-05-08T02:10:53Z', u'Nautical twilight')
(u'2017-05-08T03:04:40Z', u'Civil twilight')
(u'2017-05-08T03:49:08Z', u'\u2191 5:51')
(u'2017-05-08T11:28:23Z', u'\u2600\ufe0f 13:28')
(u'2017-05-08T19:03:59Z', u'\u2193 21:06')
(u'2017-05-08T19:07:39Z', u'Civil twilight')
(u'2017-05-08T19:52:06Z', u'Nautical twilight')
(u'2017-05-08T20:45:53Z', u'Astronomical twilight')
(u'2017-05-09T00:51:56Z', u'Astronomical twilight')
(u'2017-05-09T02:08:25Z', u'Nautical twilight')
(u'2017-05-09T03:02:45Z', u'Civil twilight')
(u'2017-05-09T03:47:29Z', u'\u2191 5:49')
(u'2017-05-09T11:28:21Z', u'\u2600\ufe0f 13:28')

```

**Comment by four2six** [comment]
no idea how to get that consistently from my machine to an editing process and strip that of personal data... etc

**Comment by allan-gam** [comment]
OK. I will try to solve it here then.

**Comment by four2six** [comment]
sorry for being difficult, i guess
if there's a way we can get you to ssh into my machine for that purpose, we can do that. not sure how to  send you the details, though.
and i'd have to open the port, i think.

**Comment by allan-gam** [comment]
Maybe you can copy a single line from the json file here. I'm particularly interested in the summary line. Below is an example from one of my files:

```
[
  {
    "created": "2016-08-29T14:11:19.000Z", 
    "creator": {
      "email": "xxxxx@gmail.com"
    }, 
    "description": "Schemalagd lektion.", 
    "end": {
      "dateTime": "2017-05-09T13:55:00+02:00", 
      "timeZone": "Europe/Stockholm"
    }, 
    "etag": "\"2944968029150000\"", 
    "htmlLink": "https://www.google.com/calendar/event?eid=OGU0NzNXXX", 
    "iCalUID": "8e473mh69lv2oskf2XXXXXbdc@google.com", 
    "id": "8e473mh69lv2oskf2v6ead7bdc_20170509T105500Z", 
    "kind": "calendar#event", 
    "location": "398", 
    "organizer": {
      "displayName": "Some school", 
      "email": "qXXXXXXXXf0t20@group.calendar.google.com", 
      "self": true
    }, 
    "originalStartTime": {
      "dateTime": "2017-05-09T12:55:00+02:00", 
      "timeZone": "Europe/Stockholm"
    }, 
    "recurringEventId": "8e47XXXXskf2v6ead7bdc", 
    "reminders": {
      "useDefault": true
    }, 
    "sequence": 0, 
    "start": {
      "dateTime": "2017-05-09T12:55:00+02:00", 
      "timeZone": "Europe/Stockholm"
    }, 
    "status": "confirmed", 
    "summary": "Matematik BGT", 
    "updated": "2016-08-29T15:20:14.575Z"
  }
]
```

**Comment by allan-gam** [comment]
And I'm particularly interested in the summary line for the current event (if any) or the upcoming event.

**Comment by four2six** [comment]
"summary": "↓ 21:04",
that's what the next event has as a summary, see also above for the whole event

**Comment by four2six** [comment]
http://www.webcal.fi/cal.php?id=42&format=ics&locationname=Solingen&lat=51.14160957755973&lon=7.016658782958984&ss=1&sn=1&ci=1&na=1&as=1&eq=1&heli=1&wp=1&wf=4&du=km&color=%23FFD966&cntr=&lang=en&rid=wc

this should recreate the calendar when you add it to yours

**Comment by allan-gam** [comment]
Unfortunately it works here.

Can you edit your /var/tmp/GCal/XXXXXXX@group.calendar.google.com.json file and see what entry that causes the error? (The original file will be recreated later if you delete it)

My idea is to find exactly what entry gives the problem. If you can isolate it to one entry, try to delete any suspect characters ...

**Comment by four2six** [comment]
that'll take some time

**Comment by four2six** [comment]
would trying to use keywords to limit the imported events help?

**Comment by four2six** [comment]
obviously not.

**Comment by allan-gam** [comment]
I have recreated such a calendar here. It works fine here unfortunately. It's something specific to your system.

Currently I have no idea what I shall do. Best option is if you can try to help me debugging it on your machine. Notepad++ is a great editor if you have Windows. It can connect to your Raspberry Pi using ssh.

**Comment by allan-gam** [comment]
I will have to quit work for today. Back again tomorrow. Cheers!

**Comment by four2six** [comment]
cheers, i'll think about it, but i guess i'm out of this.

**Comment by four2six** [comment]
i found a way to edit the file and save it back to the machine. an no, not windows, osx. ;)
what i did:
i took the first event from the xxx@import.calendar.google.com.json and made that the only event in the file. after that i ran /usr/bin/python /home/pi/GCal/gcal.py -v -d
then i did the same with the second event.
both produced the same `Unexpected response from Domoticz: ERR Url: http://192.168.2.220:8080/json.htm?vtype=2&type=command&param=updateuservariable&vname=GCal-Sun+Events-trippedEvent&vvalue=`
- is this the way you wanted that to be done for all events?
- you keep calling it /var/tmp/GCal/XXXXXXX@group.calendar.google.com.json, but for me it's /var/tmp/GCalData/XXXXXXX@group.calendar.google.com.json. Is this correct?

**Comment by allan-gam** [comment]
You are right about the name. GCalData is correct.

Your results are interesting. What if you edit the file so it has no entries at all, like this:
```
[]
```

Is it correct that you choose 'Sun Events' as Calendar Short name?

If you open the following address in a browser, what will the response be?
http://192.168.2.220:8080/json.htm?vtype=2&type=command&param=updateuservariable&vname=GCal-Sun+Events-trippedEvent&vvalue=Hello

Please check that you have a variable named 'GCal-Sun Events-trippedEvent'

Will continue with this tomorrow.

**Comment by allan-gam** [comment]
Two more questions:

When installing, did you run `sudo apt-get update && sudo apt-get dist-upgrade` ?

What is your Domoticz version?

**Comment by four2six** [comment]
response is 

```
{
   "status" : "OK",
   "title" : "UpdateUserVariable"
}
```

'Sun Events' was the Calendar Short name.

if sudo apt-get update && sudo apt-get dist-upgrade was in your instructions, i ran it.

```
About Domoticz
Version: 3.5877
Build Hash: 15b013c
Compile Date: 2016-11-10 16:44:33
```

**Comment by four2six** [comment]
'GCal-Sun Events-trippedEvent' is existant, and now at 22:57 my time it has the value of Astonomical twilight, which is the actual event.

`GCal-Sun Events-trippedEvent	Zeichenkette	Astronomical twilight	2017-05-08 22:46:14`


```
Debug is on
Google Calendar API for Domoticz Version 0.2.4 (DB version 1.0.1) running on TTY console...
Found a total of : 3 GCal configuration entries.

Syncing Calendar data with Google: xxx@import.calendar.google.com. Last fetched: 67 minutes ago. Interval: 5
Getting events for calendar xxx@import.calendar.google.com
(u'2017-05-08T00:56:25Z', u'Astronomical twilight')
(u'2017-05-08T02:10:53Z', u'Nautical twilight')
(u'2017-05-08T03:04:40Z', u'Civil twilight')
(u'2017-05-08T03:49:08Z', u'\u2191 5:51')
(u'2017-05-08T11:28:23Z', u'\u2600\ufe0f 13:28')
(u'2017-05-08T19:03:59Z', u'\u2193 21:06')
(u'2017-05-08T19:07:39Z', u'Civil twilight')
(u'2017-05-08T19:52:06Z', u'Nautical twilight')
(u'2017-05-08T20:45:53Z', u'Astronomical twilight')
(u'2017-05-09T00:51:56Z', u'Astronomical twilight')
(u'2017-05-09T02:08:25Z', u'Nautical twilight')
(u'2017-05-09T03:02:45Z', u'Civil twilight')
(u'2017-05-09T03:47:29Z', u'\u2191 5:49')
(u'2017-05-09T11:28:21Z', u'\u2600\ufe0f 13:28')
(u'2017-05-09T19:05:31Z', u'\u2193 21:07')
(u'2017-05-09T19:09:12Z', u'Civil twilight')
(u'2017-05-09T19:53:56Z', u'Nautical twilight')
(u'2017-05-09T20:48:16Z', u'Astronomical twilight')
(u'2017-05-10T00:47:18Z', u'Astronomical twilight')
(u'2017-05-10T02:05:58Z', u'Nautical twilight')
User variable: GCal-Sun Events-eventsToday
N: 9
D: 9

User variable: GCal-Sun Events-remainingEventsToday
N: 1
D: 1

User variable: GCal-Sun Events-trippedEvent
N: Astronomical twilight
D: Astronomical twilight

Failure getting data for domoticz device idx: 199
Failure getting data for domoticz device idx: 198
Google Calendar data is up to date: xxx@import.calendar.google.com. Last fetched: 0 minutes ago. Interval: 5
User variable: GCal-test 2-eventsToday
N: 2
D: 2

User variable: GCal-test 2-remainingEventsToday
N: 0
D: 0

User variable: GCal-test 2-trippedEvent
N: 
D: Civil twilight

Unexpected response from Domoticz: ERR Url: http://192.168.2.220:8080/json.htm?vtype=2&type=command&param=updateuservariable&vname=GCal-test+2-trippedEvent&vvalue=
```

**Comment by four2six** [comment]
if i missed any of your questions, let me know.

**Comment by allan-gam** [comment]
I cant use your test because when you ran using `[]` is the json I see that it started to download the file immediately hence destroying your changes by adding all the entries again. Please put a higher interval first (5 is very low for a non calendar that is not expected to be updated continously, so use 10080) After changing that, please run it again. Meanwhile I will continue to look here if I can do more.

**Comment by four2six** [comment]
the result when [] is the json with 10080 interval:

```
Debug is on
Google Calendar API for Domoticz Version 0.2.4 (DB version 1.0.1) running on TTY console...
Found a total of : 1 GCal configuration entries.

Syncing Calendar data with Google:xxx@import.calendar.google.com. Last fetched: 14400 minutes ago. Interval: 10080
Getting events for calendar xxx@import.calendar.google.com
(u'2017-05-08T11:28:23Z', u'\u2600\ufe0f 13:28')
(u'2017-05-08T19:03:59Z', u'\u2193 21:06')
(u'2017-05-08T19:07:39Z', u'Civil twilight')
(u'2017-05-08T19:52:06Z', u'Nautical twilight')
(u'2017-05-08T20:45:53Z', u'Astronomical twilight')
(u'2017-05-09T00:51:56Z', u'Astronomical twilight')
(u'2017-05-09T02:08:25Z', u'Nautical twilight')
(u'2017-05-09T03:02:45Z', u'Civil twilight')
(u'2017-05-09T03:47:29Z', u'\u2191 5:49')
(u'2017-05-09T11:28:21Z', u'\u2600\ufe0f 13:28')
(u'2017-05-09T19:05:31Z', u'\u2193 21:07')
(u'2017-05-09T19:09:12Z', u'Civil twilight')
(u'2017-05-09T19:53:56Z', u'Nautical twilight')
(u'2017-05-09T20:48:16Z', u'Astronomical twilight')
(u'2017-05-10T00:47:18Z', u'Astronomical twilight')
(u'2017-05-10T02:05:58Z', u'Nautical twilight')
(u'2017-05-10T03:00:52Z', u'Civil twilight')
(u'2017-05-10T03:45:52Z', u'\u2191 5:48')
(u'2017-05-10T11:28:19Z', u'\u2600\ufe0f 13:28')
User variable: GCal-Sun Events-eventsToday
N: 9
D: 9

User variable: GCal-Sun Events-remainingEventsToday
N: 5
D: 5

User variable: GCal-Sun Events-trippedEvent
N: 
D: 0

Unexpected response from Domoticz: ERR Url: http://192.168.2.220:8080/json.htm?vtype=2&type=command&param=updateuservariable&vname=GCal-Sun+Events-trippedEvent&vvalue=
```

i deleted all other jsons and all other switches i had to make it more of a clean install, hence the difference configuration entries. other than that, it's the same as above.

**Comment by four2six** [comment]
all devices and user variables exist as they should, the text device and the GCal-Sun Events-trippedEvent are empty or zero.

**Comment by allan-gam** [comment]
Thanks. Doing it that way, you also removed the info for when the file was latest downloaded and the events for calendar xxx@import.calendar.google.com was downloaded again, hence overwriting your changes again. Can You edit the json file again and not change anything else please. Thanks ;)

**Comment by four2six** [comment]
not sure what you want me to do.

**Comment by allan-gam** [comment]
Please edit the file so it has no entries at all, like this:
```
[]
```
Then run 
```
/usr/bin/python /home/pi/GCal/gcal.py -d -v
```

**Comment by allan-gam** [comment]
Also, I updated to 0.2.5 that will put some more debugging info. It might be helpful.

**Comment by four2six** [comment]
that's what i did. i did it again, a last time, and the result is the same.

**Comment by four2six** [comment]
after updating,
the end is with the [] still in /var/tmp/GCalData/XXXXXXX@group.calendar.google.com.json

Unexpected response from Domoticz: ERR Url: http://192.168.2.220:8080/json.htm?vtype=2&type=command&param=updateuservariable&vname=GCal-Sun+Events-trippedEvent&vvalue=
{'vtype': 2, 'type': 'command', 'vname': 'GCal-Sun Events-trippedEvent', 'param': 'updateuservariable', 'vvalue': ''}

**Comment by allan-gam** [comment]
What type is your user variable named GCal-Sun Events-trippedEvent ? Current value is 0?

**Comment by four2six** [comment]
![bildschirmfoto 2017-05-09 um 09 39 53](https://cloud.githubusercontent.com/assets/4852434/25840287/877ca43a-349b-11e7-9628-ec18bb5a4ffd.png)

**Comment by allan-gam** [comment]
This time, omitting the vvalue value at the end ... 

What if you open this is a browser:
`http://192.168.2.220:8080/json.htm?vtype=2&type=command&param=updateuservariable&vname=GCal-Sun+Events-trippedEvent&vvalue=`

**Comment by four2six** [comment]
```
{
   "status" : "ERR"
}
```

**Comment by allan-gam** [comment]
That's great news! Then we have something to work with.
Let me think a minute...

**Comment by allan-gam** [comment]
I have a newer version of Domoticz. Things are not working the same.

Please download the latest version.

It will put an "X" in the GCal-Sun+Events-trippedEvent user variable instead of a blank value. Does that work? Putting an X is just a test to make sure.

**Comment by allan-gam** [comment]
Changed in 0.2.7
The text "No event" will be used to set Domoticz user variables when there is no tripped event.

**Comment by four2six** [comment]
![bildschirmfoto 2017-05-09 um 10 29 24](https://cloud.githubusercontent.com/assets/4852434/25841949/6ea3476e-34a2-11e7-8ae9-a769be6a28e5.png)

**Comment by allan-gam** [comment]
Seems to work now?

**Comment by four2six** [comment]
the text device is correct. not sure what 'tripped event' is supposed to mean... and i don't see the x. so not knowing what is supposed to happen, you be the judge.

**Comment by allan-gam** [comment]
The GCal-XXXX-trippedEvent user variable will hold the name of the tripped event (if any).

Thank you for helping to swat this nasty bug. I couldn't have done it without your kind help because my Domoticz version is not the same as yours and is apparently not working in the same way. (Mine allows setting the user variable to a blank value using the API while yours doesn't) 

Have a nice day and once again, thanks!

**Comment by four2six** [comment]
thanks for your work, you had the most thinking to do :-)

you mean 'triggered'? 'tripped' in my knowledge means dropped, broken, failed, etc. it can mean a tripped fuse, so the fuse is set off for protection... but i guess you really mean triggered. a triggered event would be the one, in our case, when the switch device is turned on is the reason for that. so the variable would carry the name of the event that caused the switch to be turned on, to be triggered.

now i have to do a new installation of domoticz on another raspberry, as the sd card on that failed. i should think about tmpfs, even though i can't really find any proof that it really extends sd card life with domoticz.

**Comment by allan-gam** [comment]
I've used the term tripped for "to make a switch go on or off" http://www.macmillandictionary.com/dictionary/british/trip_2 It might be better to use triggered as you propose but I'd like to keep 'tripped' for now so we don't have to make a new major upgrade just for that. 

I'd buy [this](http://www.ebay.de/sch/i.html?_from=R40&_trksid=p2380057.m570.l1313.TR0.TRC0.H0.XSanDisk+Extreme+Pro+MicroSDHC+UHS-I+U3+95MB%2Fs+32GB.TRS0&_nkw=SanDisk+Extreme+Pro+MicroSDHC+UHS-I+U3+95MB%2Fs+32GB&_sacat=0). "No name"  and other cheap crap variants should be avoided.

Cheers

**Comment by four2six** [comment]
but i took your advice and ordered that one... so thanks, now the chickens have no more domoticz until that arrives ;-)

**Comment by allan-gam** [comment]
I kept some before because it's so nice to be surrounded by poultry in the garden.

**Comment by four2six** [comment]
when they don't dig it all up ;-)
